lib/linux_admin/dns.rb

```diff
@@ -0,0 +1,38 @@
+module LinuxAdmin
+  class Dns
+    attr_accessor :filename
+    attr_accessor :nameservers
+    attr_accessor :search_order
+
+    def initialize(filename = "/etc/resolv.conf")
+      @filename = filename
+      reload
+    end
+
+    def reload
+      @search_order = []
+      @nameservers  = []
+      lines = File.read(@filename).split("\n")
+      lines.each do |line|
+        if line =~ /^search .*/
+          add_search(line)
+        elsif line =~ /^nameserver .*/
+          @nameservers.push(line.split[1])
+        end
+      end
+    end
+
+    def save
+      contents = "search #{@search_order.join(" ")}\n"
+      @nameservers.each { |ns| contents += "nameserver #{ns}\n" }
+      File.write(@filename, contents)
+    end
+
+    private
+
+    def add_search(line)
```

Comment:
Do you think this would be cleaner?
```ruby
@search_order += line.partition(/^search\s+/)[2].split
```

If that's the case, it could live on line 18.